### PR TITLE
[Pygen] Fix minor issues.

### PR DESCRIPTION
**Pull request by aneels3**
This PR adds the conversion of all the string comparison of riscv_core_setting fields to a python enum type.
This fix was needed in order to implement the previous TODO's related to processor configuration.

### CODE REVIEW

**Comment by ishita71** [comment]
@googlebot I consent

File: pygen/pygen_src/riscv_asm_program_gen.py [hunk 1]
```diff
@@ -125,15 +126,16 @@ def gen_program(self):
                 if not cfg.no_data_page:
                     self.gen_data_page(hart)
 
-                    if((hart == 0) and ("RV32A" in rcs.supported_isa)):
+                    if((hart == 0) and (riscv_instr_group_t.RV32A
+                                         in rcs.supported_isa)):
```
Comment (hodjat91) [inline pygen/pygen_src/riscv_asm_program_gen.py:130]: nit: redundant parenthesis: 
`if(hart == 0 and riscv_instr_group_t.RV32A
                                         in rcs.supported_isa):`

File: pygen/pygen_src/riscv_asm_program_gen.py [hunk 2]
```diff
@@ -269,39 +271,39 @@ def setup_misa(self):
...
+            elif group in [riscv_instr_group_t.RVV]:
                 misa[misa_ext_t.MISA_EXT_V] = 1
-            elif rcs.supported_isa[i] in [riscv_instr_group_t.RV32X.name,
-                                          riscv_instr_group_t.RV64X.name]:
+            elif group in [riscv_instr_group_t.RV32X,
+                           riscv_instr_group_t.RV64X]:
                 misa[misa_ext_t.MISA_EXT_X] = 1
             else:
-                logging.error("{} is not yet supported".format(rcs.supported_isa[i]))
+                logging.error("{} is not yet supported".format(group.name))
```
Comment (hodjat91) [inline pygen/pygen_src/riscv_asm_program_gen.py:306]: Maybe add a sys.exit() after the error?

File: pygen/pygen_src/riscv_asm_program_gen.py [hunk 3]
```diff
@@ -609,48 +612,46 @@ def gen_plic_section(self, interrupt_handler_instr):
...
                 interrupt_handler_instr.append("csrsi 0x%0x, 0x%0x" % (status.value, 8))
-            elif(status == privileged_reg_t.SSTATUS):
+            elif status == privileged_reg_t.SSTATUS:
                 interrupt_handler_instr.append("csrsi 0x%0x, 0x%0x" % (status.value, 2))
-            elif(status == privileged_reg_t.USTATUS):
+            elif status == privileged_reg_t.USTATUS:
                 interrupt_handler_instr.append("csrsi 0x%0x, 0x%0x" % (status.value, 1))
             else:
-                logging.critical("Unsupported status %0s" % (status.value))
+                logging.critical("Unsupported status {}".format(status.name))
```
Comment (hodjat91) [inline pygen/pygen_src/riscv_asm_program_gen.py:654]: Please add sys.exit after the logging.critical.

File: pygen/pygen_src/riscv_asm_program_gen.py [hunk 4]
```diff
@@ -734,24 +735,24 @@ def generate_directed_instr_stream(self, hart = 0, label = "", original_instr_cn
...
+            if instr_insert_cnt <= min_insert_cnt:
                 instr_insert_cnt = min_insert_cnt
             logging.info("Insert directed instr stream %0s %0d/%0d times",
                          instr_stream_name, instr_insert_cnt, original_instr_cnt)
             for i in range(instr_insert_cnt):
                 name = "{}_{}".format(instr_stream_name, i)
                 object_h = factory(instr_stream_name)
                 object_h.name = name
-                if(object_h is None):
+                if object_h is None:
```
Comment (hodjat91) [inline pygen/pygen_src/riscv_asm_program_gen.py:751]: nit: `if not object_h:`

File: pygen/pygen_src/riscv_instr_pkg.py [hunk 5]
```diff
@@ -1495,10 +1495,10 @@ def push_gpr_to_kernel_stack(self, status, scratch, mprv, sp, tp, instr):
         # If MPRV is set and MPP is S/U mode, it means the address translation and
         # memory protection for load/store instruction is the same as the mode indicated
         # by MPP. In this case, we need to use the virtual address to access the kernel stack.
-        if((status.name == "MSTATUS") and (rcs.SATP_MODE != "BARE")):
+        if((status.name == privileged_reg_t.MSTATUS) and (rcs.SATP_MODE != satp_mode_t.BARE)):
```
Comment (hodjat91) [inline pygen/pygen_src/riscv_instr_pkg.py:1498]: nit: redundant parenthesis.

**Review by hodjat91 — APPROVED** [review]
LGTM! Thanks!

**Review by udinator — APPROVED** [review]

**Pull request merged**